crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -28,21 +28,29 @@ pub struct TransactionReceipt<T = ReceiptEnvelope<Log>> {
     #[serde(with = "alloy_serde::u64_hex_opt")]
     pub block_number: Option<u64>,
     /// Gas used by this transaction alone.
-    #[serde(with = "alloy_serde::u64_hex_opt")]
-    pub gas_used: Option<u64>,
+    #[serde(with = "alloy_serde::u128_hex_or_decimal_opt")]
+    pub gas_used: Option<u128>,
```

Comment:
needs `default` and `skip_serializing_if`

aren't those mandatory though?

Reply:
<img width="639" alt="image" src="https://github.com/alloy-rs/alloy/assets/62447812/b5e303c4-8909-4138-822e-81b9c60adedb">

Reply:
gas used is required:

```
// Receipt represents the results of a transaction.
type Receipt struct {
	// Consensus fields: These fields are defined by the Yellow Paper
	Type              uint8  `json:"type,omitempty"`
	PostState         []byte `json:"root"`
	Status            uint64 `json:"status"`
	CumulativeGasUsed uint64 `json:"cumulativeGasUsed" gencodec:"required"`
	Bloom             Bloom  `json:"logsBloom"         gencodec:"required"`
	Logs              []*Log `json:"logs"              gencodec:"required"`

	// Implementation fields: These fields are added by geth when processing a transaction.
	TxHash            common.Hash    `json:"transactionHash" gencodec:"required"`
	ContractAddress   common.Address `json:"contractAddress"`
	GasUsed           uint64         `json:"gasUsed" gencodec:"required"`
	EffectiveGasPrice *big.Int       `json:"effectiveGasPrice"` // required, but tag omitted for backwards compatibility
	BlobGasUsed       uint64         `json:"blobGasUsed,omitempty"`
	BlobGasPrice      *big.Int       `json:"blobGasPrice,omitempty"`

	// Inclusion information: These fields provide information about the inclusion of the
	// transaction corresponding to this receipt.
	BlockHash        common.Hash `json:"blockHash,omitempty"`
	BlockNumber      *big.Int    `json:"blockNumber,omitempty"`
	TransactionIndex uint        `json:"transactionIndex"`
}
```

Reply:
@yash-atreya